File: Baekjoon_No10866.py
import sys
from collections import deque
result = deque()
input_count = int(sys.stdin.readline())
for _ in range(input_count):
    data = list(sys.stdin.readline().split())

    if data[0] == 'push_front':
        result.appendleft(data[1])
        
    elif data[0] == 'push_back':
        result.append(data[1])
        
    elif data[0] == 'size':
        sys.stdout.write(str(len(result)))
        sys.stdout.write("\n")
        
    elif data[0] == 'pop_front':    # pop이 불가할 경우 -1로 예외처리
        if len(result) == 0:
            sys.stdout.write("-1\n")
        else:
            sys.stdout.write(str(result.popleft()))
            sys.stdout.write("\n")
            
    elif data[0] == 'pop_back':    # pop이 불가할 경우 -1로 예외처리
        if len(result) == 0:
            sys.stdout.write("-1\n")
        else:
            sys.stdout.write(str(result.pop()))
            sys.stdout.write("\n")
            
    elif data[0] == 'back':
        if len(result) == 0:
            sys.stdout.write("-1\n")
        else:
            sys.stdout.write(str(result[len(result)-1]))
            sys.stdout.write("\n")

    elif data[0] == 'front':
        if len(result) == 0:
            sys.stdout.write("-1\n")
        else:
            sys.stdout.write(str(result[0]))
            sys.stdout.write("\n")

    elif data[0] == 'empty':
        if len(result) == 0:
            sys.stdout.write("1\n")
        else:
            sys.stdout.write("0\n")
            
            

File: 10000/Baekjoon_No10866.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class b10866 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        Deque<Integer> deque = new ArrayDeque<>();

        int n = Integer.parseInt(br.readLine());

        for (int i = 0; i < n; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine(), " ");
            String first = st.nextToken();

            if (st.hasMoreTokens()) {
                int second = Integer.parseInt(st.nextToken());

                if (first.equals("push_front")) {
                    deque.addFirst(second);
                }

                else if (first.equals("push_back")) {
                    deque.addLast(second);
                }
            }

            else if (first.equals("pop_front")) {
                if (deque.isEmpty()) {
                    System.out.println("-1");
                }
                else {
                    System.out.println(deque.pollFirst());
                }
            }

            else if (first.equals("pop_back")) {
                if (deque.isEmpty()) {
                    System.out.println("-1");
                }
                else {
                    System.out.println(deque.pollLast());
                }
            }

            else if (first.equals("front")) {
                if (deque.isEmpty()) {
                    System.out.println("-1");
                }
                else {
                    System.out.println(deque.peekFirst());
                }
            }

            else if (first.equals("back")) {
                if (deque.isEmpty()) {
                    System.out.println("-1");
                }
                else {
                    System.out.println(deque.peekLast());
                }
            }
            
            else if (first.equals("size")) {
                System.out.println(deque.size());
            }

            else if (first.equals("empty")) {
                if (deque.isEmpty()) {
                    System.out.println("1");
                }
                else {
                    System.out.println("0");
                }
            }
        }
    }
}
